proto/redpanda/api/dataplane/v1/security.proto

```diff
@@ -0,0 +1,370 @@
+      summary: "Create Role"
+      description: "Create a role."
+      responses: {
+        key: "201"
+        value: {
+          description: "Role created"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.CreateRoleResponse"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_ADMIN
+      api: API_REDPANDA_ADMIN
+    };
+  }
+
+  // GetRole retrieves the specific role.
+  rpc GetRole(GetRoleRequest) returns (GetRoleResponse) {
+    option (google.api.http) = {get: "/v1/roles/{role_name}"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Get Role"
+      description: "Get a role."
+      responses: {
+        key: "200"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.GetRoleResponse"}
+          }
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_VIEW
+      api: API_REDPANDA_ADMIN
+    };
+  }
+
+  // DeleteRole deletes the role from the system.
+  rpc DeleteRole(DeleteRoleRequest) returns (DeleteRoleResponse) {
+    option (google.api.http) = {delete: "/v1/roles/{role_name}"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Delete Role"
+      description: "Delete a Role."
+      responses: {
+        key: "204"
+        value: {
+          description: "Role deleted successfully"
+          schema: {}
+        }
+      }
+      responses: {
+        key: "404"
+        value: {
+          description: "Not Found"
+          schema: {
+            json_schema: {ref: ".google.rpc.Status"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_ADMIN
+      api: API_REDPANDA_ADMIN
+    };
+  }
+
+  // ListRoleMembership lists all the members assigned to a role based on optional filter.
+  rpc ListRoleMembers(ListRoleMembersRequest) returns (ListRoleMembersResponse) {
+    option (google.api.http) = {get: "/v1/roles/{role_name}/members"};
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "List Role Members"
+      description: "List members assigned to a role."
+      responses: {
+        key: "200"
+        value: {
+          description: "OK"
+          schema: {
+            json_schema: {ref: ".redpanda.api.dataplane.v1.ListRoleMembersResponse"}
+          }
+        }
+      }
+    };
+    option (redpanda.api.auth.v1.authorization) = {
+      required_permission: PERMISSION_VIEW
+      api: API_REDPANDA_ADMIN
+    };
+  }
```

Comment:
Should `ListRoles`, `GetRole`, and `ListRoleMembers` be available for viewer permissions?

Reply:
Changed all permissions to be PERMISSION_ADMIN